Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -470,6 +470,66 @@ void Node::invalidate_style(StyleInvalidationReason reason)
     document().schedule_style_update();
 }
 
+void Node::invalidate_style(StyleInvalidationReason reason, Vector<CSS::InvalidationSet::Property> const& properties)
+{
+    if (is_character_data())
+        return;
+
+    bool properties_used_in_has_selectors = false;
+    for (auto const& property : properties) {
+        if (document().style_computer().class_name_used_in_has_selector(property.name)) {
+            properties_used_in_has_selectors = true;
+            break;
+        }
+    }
```

Comment:
Shouldn't this look check the `property.type` as well?

Reply:
indeed, it should! it has only worked fine because so far it's only invoked for class mutations. I added the check.